app/main.py

```diff
@@ -1,3 +1,95 @@
-def shop_trip():
-    # write your code here
-    pass
+import math
+import datetime
+from app.customer import CustomerCar
+from app.shop import Shop
+from app.customer import customers_and_content
+
+
+def shop_trip() -> str:
+    customers, content, shops, priсe_fuel = customers_and_content()
+    for name_customer in customers:
+        name = name_customer.get("name")
+        product_cart = name_customer.get("product_cart")
+        location = name_customer.get("location")
+        money = name_customer.get("money")
+        car = name_customer.get("car")
+        customer_instance = (
+            CustomerCar(name, product_cart, location, money, car)
+        )
+        customer_inf = customer_instance.customer_location()
+        fuel_consumption_car = customer_inf["fuel_consumption_car"]
+        distance_customer_x = customer_inf["distance_customer_x"]
+        distance_customer_y = customer_inf["distance_customer_y"]
+        money = customer_inf["money"]
+        product_cart = customer_inf["product_cart"]
+        name = customer_inf["name"]
```

Comment:
This lines is redundant -> 
fuel_consumption_car -> customer_instance.car.fuel_consumption
name -> customer_instance.name

Reply:
And others